packages/core/src/actions/send.ts

```diff
@@ -99,23 +99,39 @@ export function send<
           : params.delay;
       }
 
-      let resolvedTarget = isFunction(params.to)
+      const resolvedTarget = isFunction(params.to)
         ? params.to(ctx, _event.data, meta)
         : params.to;
-      resolvedTarget =
-        isString(resolvedTarget) &&
-        resolvedTarget !== SpecialTargets.Parent &&
-        resolvedTarget !== SpecialTargets.Internal &&
-        resolvedTarget.startsWith('#_')
-          ? resolvedTarget.slice(2)
-          : resolvedTarget;
+      let targetActorRef: AnyActorRef | undefined;
+
+      if (typeof resolvedTarget === 'string') {
+        if (resolvedTarget === SpecialTargets.Parent) {
+          targetActorRef = actorContext?.self._parent;
+        } else if (
+          resolvedTarget === SpecialTargets.Internal ||
+          // SCXML compatibility
+          resolvedTarget === '#_internal'
+        ) {
+          targetActorRef = actorContext?.self;
+        } else if (resolvedTarget?.startsWith('#_')) {
+          // SCXML compatibility
+          targetActorRef = state.children[resolvedTarget.slice(2)];
+        } else {
+          targetActorRef = state.children[resolvedTarget];
+          if (!targetActorRef) {
+            throw new Error('no');
+          }
+        }
+      } else {
+        targetActorRef = resolvedTarget;
```

Comment:
There is a weird interplay between this resolving here and the call site that is supposed to execute things:
https://github.com/statelyai/xstate/blob/907288c688c6179db2ed09efd66e5ad0d5ebbd21/packages/core/src/exec.ts#L110-L115

I think we could simplify this a little bit If we'd do this here:
```suggestion
        targetActorRef = actorContext?.self;
```

I also wonder why the send to self is not "deferred" while the send to other actors is. They probably could be handled in the same way.

Reply:
> I also wonder why the send to self is not "deferred" while the send to other actors is. They probably could be handled in the same way.

I believe there are some tests that fail if we try to handle it the same way; will look again

Reply:
Just checked; this causes many tests to fail.